### PR TITLE
feat(api): add bindings for imgui-knobs

### PR DESCRIPTION
**Pull request by 1312Delta**
Add bindings for the imgui-knobs extension

# Description

Add jni bindings for the imgui-knobs extension and example window for it

## Type of change

- [ ] Minor changes or tweaks (quality of life stuff)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -33,3 +33,5 @@ logs/
 build/
 out/
 !gradle/wrapper/*.jar
+
+gradle.properties
```
Comment (SpaiR) [inline .gitignore:37]: Can you describe, why was this added to the gitignore?

File: example/src/main/java/Extra.java [hunk 2]
```diff
@@ -30,6 +31,7 @@ public static void show(final Application app) {
         ImGui.checkbox("Show ImGui MemoryEditor Demo Window", SHOW_IMGUI_MEMORY_EDITOR_WINDOW);
         ImGui.checkbox("Show ImGui Canvas Demo Window", SHOW_IMGUI_CANVAS_EDITOR_WINDOW);
         ImGui.checkbox("Show Imgui InputText Callback Window", SHOW_IMGUI_INPUT_CALLBACK_WINDOW);
+        ImGui.checkbox("Show Imgui Knobs Demo",SHOW_IMGUI_KNOBS_DEMO);
```
Comment (SpaiR) [inline example/src/main/java/Extra.java:34]:
```suggestion
        ImGui.checkbox("Show Imgui Knobs Demo", SHOW_IMGUI_KNOBS_DEMO);
```

File: example/src/main/java/Extra.java [hunk 3]
```diff
@@ -74,5 +76,9 @@ public static void show(final Application app) {
         if (SHOW_IMGUI_INPUT_CALLBACK_WINDOW.get()) {
             ExampleInputTextCallback.show(SHOW_IMGUI_INPUT_CALLBACK_WINDOW);
         }
+
+        if(SHOW_IMGUI_KNOBS_DEMO.get()){
```
Comment (SpaiR) [inline example/src/main/java/Extra.java:80]:
```suggestion
        if (SHOW_IMGUI_KNOBS_DEMO.get()) {
```

File: example/src/main/java/ExampleKnobs.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+public class ExampleKnobs {
+
+    private static final ImFloat exampleTickKnobVal = new ImFloat(0);
+    private static final ImFloat exampleDotKnobVal = new ImFloat(0);
+    private static final ImFloat exampleSpaceKnobVal = new ImFloat(0);
+    private static final ImFloat exampleWiperKnobVal = new ImFloat(0);
+    private static final ImFloat exampleWiperDotKnobVal = new ImFloat(0);
+    private static final ImFloat exampleWiperOnlyKnobVal = new ImFloat(0);
+    private static final ImFloat exampleSteppedKnobVal = new ImFloat(0);
```
Comment (SpaiR) [inline example/src/main/java/ExampleKnobs.java:18]: Please convert field names to UPPER_CASE.

File: imgui-binding/src/main/java/imgui/extension/imguiknobs/flag/ImGuiKnobFlags.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package imgui.extension.imguiknobs.flag;
+
+public final class ImGuiKnobFlags {
```
Comment (SpaiR) [inline imgui-binding/src/main/java/imgui/extension/imguiknobs/flag/ImGuiKnobFlags.java:3]: May I ask you to convert those flags to the general style? For example: https://github.com/SpaiR/imgui-java/blob/main/imgui-binding/src/main/java/imgui/extension/imguizmo/flag/Operation.java

- Fields in `UPPER_CASE`
- Private constructors without body (no need to throw exception

**Review by SpaiR — CHANGES_REQUESTED** [review]
@1312Delta Looks good 👍 , but can you address styling issues I've mentioned?

File: .gitignore (hunk 1 above)
Comment (SpaiR) [reply .gitignore:37]: It's a common Gradle file that, by convention, should be available in VCS. Please remove it from .gitignore. To resolve your problem, you can set up gradle.properties locally. You can place it in the Gradle user home, which is defined by the GRADLE_USER_HOME environment variable. If it's not set, it defaults to USER_HOME/.gradle. Create a file named gradle.properties there and fill it with the data you want.

Comment (1312Delta) [reply .gitignore]: Resolved in [6909d92](https://github.com/SpaiR/imgui-java/pull/211/commits/6909d92c1d9f85fee43a73e5f6a587d3fb9896b1)

**Review by THCFree — APPROVED** [review]

**Review by THCFree — APPROVED** [review]

**Review by SpaiR — APPROVED** [review]
